bigtable-hbase-mirroring-client-1.x-parent/bigtable-hbase-mirroring-client-1.x/src/main/java/com/google/cloud/bigtable/mirroring/hbase1_x/MirroringConnection.java

```diff
@@ -0,0 +1,285 @@
+import com.google.cloud.bigtable.mirroring.hbase1_x.utils.faillog.Logger;
+import com.google.cloud.bigtable.mirroring.hbase1_x.utils.faillog.Serializer;
+import com.google.cloud.bigtable.mirroring.hbase1_x.utils.flowcontrol.FlowControlStrategy;
+import com.google.cloud.bigtable.mirroring.hbase1_x.utils.flowcontrol.FlowController;
+import com.google.cloud.bigtable.mirroring.hbase1_x.utils.mirroringmetrics.MirroringSpanConstants.HBaseOperation;
+import com.google.cloud.bigtable.mirroring.hbase1_x.utils.mirroringmetrics.MirroringTracer;
+import com.google.cloud.bigtable.mirroring.hbase1_x.utils.reflection.ReflectionConstructor;
+import com.google.cloud.bigtable.mirroring.hbase1_x.verification.MismatchDetector;
+import io.opencensus.common.Scope;
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.BufferedMutator;
+import org.apache.hadoop.hbase.client.BufferedMutatorParams;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.ConnectionFactory;
+import org.apache.hadoop.hbase.client.RegionLocator;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.security.User;
+
+public class MirroringConnection implements Connection {
+  private static final com.google.cloud.bigtable.mirroring.hbase1_x.utils.Logger Log =
+      new com.google.cloud.bigtable.mirroring.hbase1_x.utils.Logger(MirroringConnection.class);
+  private final FlowController flowController;
+  private final ExecutorService executorService;
+  private final MismatchDetector mismatchDetector;
+  private final ListenableReferenceCounter referenceCounter;
+  private final MirroringTracer mirroringTracer;
+  private final SecondaryWriteErrorConsumer secondaryWriteErrorConsumer;
+  private final ReadSampler readSampler;
+  private final Logger failedWritesLogger;
+  private final MirroringConfiguration configuration;
+  private final Connection primaryConnection;
+  private final Connection secondaryConnection;
+  private final AtomicBoolean closed = new AtomicBoolean(false);
+  private final AtomicBoolean aborted = new AtomicBoolean(false);
+  private final boolean performWritesConcurrently;
+
+  /**
+   * The constructor called from {@link
+   * org.apache.hadoop.hbase.client.ConnectionFactory#createConnection(Configuration)} and in its
+   * many forms via reflection with this specific signature.
+   *
+   * <p>Parameters are passed down to ConnectionFactory#createConnection method, connection errors
+   * are passed back to the user.
+   */
+  public MirroringConnection(Configuration conf, boolean managed, ExecutorService pool, User user)
```

Comment:
ping

Reply:
I think that all args are relevant. We are using configuration, passing through pool and user, and we want to check value of managed argument. One option would be to check managed in a separate constructor and then forward the others to private constructor, but this is not possible because `this(...)` call has to be the first statement of constructor what would reverse the order of operation - the check would be performed after creating connections with unexpected value of managed field.

Reply:
its done.